hamp_radar/serde_flight.py

```diff
@@ -38,66 +39,125 @@ def ppar_for_dataset(
     ), "Consecutive PPAR subblocks should never occuur, corrupt data?"
 
     if mmbg.count > 1:
-        # only use last PPAR for DSP configuration
         offset = mmbg.offset + mmbg.step * (mmbg.count - 1)
         count = 1
         mmbg = MultiMainBlockGeometry(
             mmbg.tag, offset, count, mmbg.step, mmbg.subblocks
         )
 
-    ppar = DatasetBlockGeometry(filename, mmbg)
+    return DatasetBlockGeometry(filename, mmbg)
+
+
+def generate_dbgs(pfgs: Iterable[PdsFileGeometry]) -> Iterable[DatasetBlockGeometry]:
+    """
+    Generates DatasetBlockGeometry intsances from a series of PdsFileGeometry instances
+
+    Args:
+        pfgs (Iterable[PdsFileGeometry]): An iterable of PdsFileGeometry instances.
+
+    Yields:
+        DatasetBlockGeometry: A DatasetBlockGeometry object for each mainblock in
+                              the PdsFileGeometry iterable.
+    """
+    for pfg in pfgs:
+        for mmbg in pfg.mainblocks:
+            if mmbg.subblocks[0].tag == "PPAR":
+                yield pparblock_for_dataset(pfg.filename, mmbg)
+            else:
+                yield DatasetBlockGeometry(pfg.filename, mmbg)
+
+
+def next_pparblock(blocks: Iterable[DatasetBlockGeometry]) -> DatasetBlockGeometry:
+    """
+    Finds and returns the next PPAR block from an (peekable) iterable of dataset blocks.
+
+    Function finds the "last of the first" PPAR block, i.e. it finds the first
+    occurance a PPAR in the list of DatasetBlockGeometry instances and if more
+    PPAR blocks follow consecutively after this one then the latest PPAR block
+    in the consecutive PPARs is returned.
+
+    Args:
+        blocks (Iterable[DatasetBlockGeometry]): Peekable iterable of dataset blocks to search through.
+
+    Returns:
+        DatasetBlockGeometry: The next PPAR block found in the iterable.
+
+    Raises:
+        IndexError: If no PPAR block is found by iterating over the dataset blocks.
+    """
+    import warnings
+
+    try:
+        while blocks.peek().mainblock.subblocks[0].tag != "PPAR":
```

Comment:
I'd suggest to define a function
```python
def is_ppar(block):
    return block.mainblock.subblocks[0].tag == "PPAR"
```
to simplify a few occurrences. Maybe the function should also be graceful in case of no subblocks.

Reply:
good idea, done :)